src/util.rs

```diff
@@ -60,49 +59,119 @@ impl Drop for Mmap {
     }
 }
 
-/// An owned file descriptor.
-pub struct Fd(pub RawFd);
+pub use fd::Fd;
 
-impl TryFrom<RawFd> for Fd {
-    type Error = ();
+#[cfg(feature = "io_safety")]
+mod fd {
+    use std::os::unix::io::{AsRawFd, RawFd, IntoRawFd, FromRawFd, OwnedFd, AsFd, BorrowedFd};
 
-    #[inline]
-    fn try_from(value: RawFd) -> Result<Fd, Self::Error> {
-        if value >= 0 {
-            Ok(Fd(value))
-        } else {
-            Err(())
+    /// An owned file descriptor.
+    pub struct Fd(pub OwnedFd);
```

Comment:
I would make `io_uring` use `OwnedFd` directly.

Reply:
What is this project's MSRV? I use the `Fd` wrapper to choose between `OwnedFd` and a version compatible with older versions of Rust. Using `OwnedFd` directly would bump the MSRV to 1.63.

Reply:
Don't have an explicit MSRV, but I really don't want to chase a compiler version that's too new.

We still have to provide a fallback impl, but the interface should be consistent with `OwnedFd`. After 1.63 is old enough, we can delete this fallback impl directly.